### PR TITLE
feat(dli): add a new data source to get quotas

### PR DESCRIPTION
**Pull request by zhangtao505**
**What this PR does / why we need it**:
add a new data source to get quotas.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
add a new data source to get quotas.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

make testacc TEST=./huaweicloud/services/acceptance/dli TESTARGS='-run TestAccDataSourceDliQuotas_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDataSourceDliQuotas_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceDliQuotas_basic
=== PAUSE TestAccDataSourceDliQuotas_basic
=== CONT  TestAccDataSourceDliQuotas_basic
--- PASS: TestAccDataSourceDliQuotas_basic (16.57s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       16.616s

### CODE REVIEW

File: docs/data-sources/dli_quotas.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+subcategory: "Data Lake Insight (DLI)"
+---
+
+# huaweicloud_dli_quotas
+
+Use this data source to get a list of quotas
```
Comment (Lance52259) [inline docs/data-sources/dli_quotas.md]:
```suggestion
Use this data source to get a list of resource quotas for DLI service within HuaweiCloud.
```

File: docs/data-sources/dli_quotas.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+```hcl
+data "huaweicloud_dli_quotas" "test" {}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the resource.
```
Comment (Lance52259) [inline docs/data-sources/dli_quotas.md:19]:
```suggestion
* `region` - (Optional, String) Specifies the region in which to query the resource quotas.
```

File: docs/data-sources/dli_quotas.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  If omitted, the provider-level region will be used.
+
+* `type` - (Optional, String) Specifies the type of quota.
```
Comment (Lance52259) [inline docs/data-sources/dli_quotas.md]:
```suggestion
* `type` - (Optional, String) Specifies the resource type that used to query corresponding quota.
```

File: docs/data-sources/dli_quotas.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  Value options:
+   + **CU**: computing unit
```
Comment (Lance52259) [inline docs/data-sources/dli_quotas.md]:
```suggestion
   + **CU**: Computing unit.
```
check the format and do not missing punctuation mark (.).

File: docs/data-sources/dli_quotas.md (hunk 1 above)
Comment (zhangtao505) [reply docs/data-sources/dli_quotas.md]: done

File: docs/data-sources/dli_quotas.md (hunk 3 above)
Comment (zhangtao505) [reply docs/data-sources/dli_quotas.md]: done

File: docs/data-sources/dli_quotas.md (hunk 4 above)
Comment (zhangtao505) [reply docs/data-sources/dli_quotas.md]: done

File: docs/data-sources/dli_quotas.md (hunk 2 above)
Comment (zhangtao505) [reply docs/data-sources/dli_quotas.md:19]: This parameter description is automatically generated by PMS.

File: huaweicloud/services/acceptance/dli/data_source_huaweicloud_dli_quotas_test.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+			},
+		},
+	})
+}
+
+func testDataSourceDataSourceDliQuotas_basic() string {
+	return `
+data "huaweicloud_dli_quotas" "test" {
+  type = ""
+}
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/dli/data_source_huaweicloud_dli_quotas_test.go]:
```suggestion
data "huaweicloud_dli_quotas" "test" {}
```

File: huaweicloud/services/acceptance/dli/data_source_huaweicloud_dli_quotas_test.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+output "no_filter_useful" {
+  value = length(data.huaweicloud_dli_quotas.test.quotas) > 1
+}
+
+output "cu_filter" {
+  value = length(data.huaweicloud_dli_quotas.cu_filter.quotas) ==1
+}
+
+output "absent_type_test" {
+  value = length(data.huaweicloud_dli_quotas.absent_type_test.quotas) ==0
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/dli/data_source_huaweicloud_dli_quotas_test.go]:
```suggestion
  value = length(data.huaweicloud_dli_quotas.cu_filter.quotas) == 1
}

output "absent_type_test" {
  value = length(data.huaweicloud_dli_quotas.absent_type_test.quotas) == 0
```

File: huaweicloud/services/acceptance/dli/data_source_huaweicloud_dli_quotas_test.go (hunk 5 above)
Comment (zhangtao505) [reply huaweicloud/services/acceptance/dli/data_source_huaweicloud_dli_quotas_test.go]: done

File: huaweicloud/services/acceptance/dli/data_source_huaweicloud_dli_quotas_test.go (hunk 6 above)
Comment (zhangtao505) [reply huaweicloud/services/acceptance/dli/data_source_huaweicloud_dli_quotas_test.go]: done

**Pull request merged**